libifupdown/state.c

```diff
@@ -16,24 +16,32 @@
 #include <string.h>
 #include "libifupdown/state.h"
 #include "libifupdown/fgetline.h"
+#include "libifupdown/tokenize.h"
 
 bool
 lif_state_read(struct lif_dict *state, FILE *fd)
 {
 	char linebuf[4096];
+
 	while (lif_fgetline(linebuf, sizeof linebuf, fd))
 	{
-		char *ifname = linebuf;
+		char *bufp = linebuf;
+		char *ifname = lif_next_token(&bufp);
+		char *refcount = lif_next_token(&bufp);
+		size_t rc = 1;
 		char *equals_p = strchr(linebuf, '=');
 
+		if (*refcount)
+			rc = strtoul(refcount, NULL, 10);
```

Comment:
Do we trust the implementation of `strtoul()` to return 0 on invalid input? According to the man page I consulted that's left to the implementation (MAY).

Reply:
i added a commit just now checking for ULONG_MAX or 0.  should be sufficient.  POSIX requires ULONG_MAX anyway.